ceno_emul/src/platform.rs

```diff
@@ -9,33 +9,42 @@ use crate::addr::{Addr, RegIdx};
 /// - codes of environment calls.
 #[derive(Clone, Debug)]
 pub struct Platform {
-    pub rom: Range<Addr>,
-    pub ram: Range<Addr>,
+    pub prog_code: Range<Addr>,
+    pub prog_data: Option<HashSet<Addr>>,
+    pub stack: Range<Addr>,
+    pub heap: Range<Addr>,
     pub public_io: Range<Addr>,
     pub hints: Range<Addr>,
-    pub stack_top: Addr,
     /// If true, ecall instructions are no-op instead of trap. Testing only.
     pub unsafe_ecall_nop: bool,
 }
 
 pub const CENO_PLATFORM: Platform = Platform {
-    rom: 0x2000_0000..0x3000_0000,
-    ram: 0x8000_0000..0xFFFF_0000,
+    prog_code: 0x2000_0000..0x3000_0000,
+    prog_data: None, // This is an `Option` to allow `const` here.
+    stack: 0xB0000000..0xC0000000,
+    heap: 0x8000_0000..0xFFFF_0000,
     public_io: 0x3000_1000..0x3000_2000,
     hints: 0x4000_0000..0x5000_0000,
-    stack_top: 0xC0000000,
     unsafe_ecall_nop: false,
 };
 
 impl Platform {
     // Virtual memory layout.
 
-    pub fn is_rom(&self, addr: Addr) -> bool {
-        self.rom.contains(&addr)
+    pub fn is_prog_code(&self, addr: Addr) -> bool {
```

Comment:
We don't need this at all in the platform.  See the comment about Harvard architecture and decoding.

Reply:
Reverted.